src/ExpressionParser.php

```diff
@@ -739,7 +759,16 @@ private function parseTestExpression(Node $node): TestExpression
             $node->setAttribute('safe', true);
         }
 
-        return new $class($node, $name, $arguments, $this->parser->getCurrentToken()->getLine());
+        $ready = $test instanceof TwigTest;
+        if (!isset($this->readyNodes[$class = $test->getNodeClass()])) {
+            $this->readyNodes[$class] = (bool) (new \ReflectionClass($class))->getConstructor()->getAttributes(FirstClassTwigCallableReady::class);
+        }
+
+        if (!$ready = $this->readyNodes[$class]) {
+            trigger_deprecation('twig/twig', '3.12', 'Twig node "%s" is not marked as ready for passing a "TwigTest" instead instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.', $class);
```

Comment:
```suggestion
            trigger_deprecation('twig/twig', '3.12', 'Twig node "%s" is not marked as ready for passing a "TwigTest" instead of its name; please make it ready and then flag its constructor with the #[FirstClassTwigCallableReady] attribute.', $class);
```